pytest_factoryboy/fixture.py

```diff
@@ -138,23 +144,35 @@ def attributes(cls, create=False, extra=None):
 
     postgen_declarations = {}
     postgen_declaration_args = {}
+    related = []
+    postgen = []
     if factory_class._meta.postgen_declarations:
         for attr, declaration in sorted(factory_class._meta.postgen_declarations.items()):
             postgen_declarations[attr] = declaration
             postgen_declaration_args[attr] = declaration.extract(attr, data)
+            if isinstance(declaration, factory.RelatedFactory):
+                related.append(attr)
+            else:
+                postgen.append(attr)
 
     result = Factory(**data)
     if postgen_declarations:
-        request._fixturedef.cached_result = (result, 0, None)
-        request._fixturedefs[request.fixturename] = request._fixturedef
+        factoryboy_request = request.getfuncargvalue("factoryboy_request")
```

Comment:
extract to separate function, the outer function became unreadable and too complex